lld/test/ELF/bp-section-orderer.s

```diff
@@ -0,0 +1,868 @@
+.Linfo_string11:
+	.asciz	"_Z1Bi"                         // string offset=70
+.Linfo_string12:
+	.asciz	"B"                             // string offset=76
+.Linfo_string13:
+	.asciz	"int"                           // string offset=78
+.Linfo_string14:
+	.asciz	"_Z1Ci"                         // string offset=82
+.Linfo_string15:
+	.asciz	"C"                             // string offset=88
+.Linfo_string16:
+	.asciz	"_Z1Di"                         // string offset=90
+.Linfo_string17:
+	.asciz	"D"                             // string offset=96
+.Linfo_string18:
+	.asciz	"_Z1Ei"                         // string offset=98
+.Linfo_string19:
+	.asciz	"E"                             // string offset=104
+.Linfo_string20:
+	.asciz	"_Z1Fi"                         // string offset=106
+.Linfo_string21:
+	.asciz	"F"                             // string offset=112
+.Linfo_string22:
+	.asciz	"main"                          // string offset=114
+.Linfo_string23:
+	.asciz	"a"                             // string offset=119
+	.section	.debug_str_offsets,"",@progbits
+	.word	.Linfo_string0
+	.word	.Linfo_string1
+	.word	.Linfo_string2
+	.word	.Linfo_string3
+	.word	.Linfo_string4
+	.word	.Linfo_string5
+	.word	.Linfo_string6
+	.word	.Linfo_string7
+	.word	.Linfo_string8
+	.word	.Linfo_string9
+	.word	.Linfo_string10
+	.word	.Linfo_string11
+	.word	.Linfo_string12
+	.word	.Linfo_string13
+	.word	.Linfo_string14
+	.word	.Linfo_string15
+	.word	.Linfo_string16
+	.word	.Linfo_string17
+	.word	.Linfo_string18
+	.word	.Linfo_string19
+	.word	.Linfo_string20
+	.word	.Linfo_string21
+	.word	.Linfo_string22
+	.word	.Linfo_string23
+	.section	.debug_addr,"",@progbits
+	.word	.Ldebug_addr_end0-.Ldebug_addr_start0 // Length of contribution
+.Ldebug_addr_start0:
+	.hword	5                               // DWARF version number
+	.byte	8                               // Address size
+	.byte	0                               // Segment selector size
+.Laddr_table_base0:
+	.xword	r1
+	.xword	r2
+	.xword	_ZL2s1
+	.xword	.Lfunc_begin0
+	.xword	.Lfunc_begin1
+	.xword	.Lfunc_begin2
+	.xword	.Lfunc_begin3
+	.xword	.Lfunc_begin4
+	.xword	.Lfunc_begin5
+	.xword	.Lfunc_begin6
+.Ldebug_addr_end0:
+	.section	".note.GNU-stack","",@progbits
+	.addrsig
+	.addrsig_sym _Z1Av
+	.addrsig_sym _Z1Bi
+	.addrsig_sym _Z1Ci
+	.addrsig_sym _ZL2s1
+	.addrsig_sym r1
+	.section	.debug_line,"",@progbits
+.Lline_table_start0:
+
+# RUN: ld.lld -e main -o a.out a.o --irpgo-profile-sort=a.profdata --verbose-bp-section-orderer 2>&1 | FileCheck %s --check-prefix=STARTUP
+# RUN: ld.lld -e main -o a.out a.o --irpgo-profile-sort=a.profdata --verbose-bp-section-orderer --icf=all --compression-sort=none 2>&1 | FileCheck %s --check-prefix=STARTUP
+
+# STARTUP: Ordered 3 sections using balanced partitioning
+
+# RUN: ld.lld -e main -o - a.o --irpgo-profile-sort=a.profdata --symbol-ordering-file a.orderfile | llvm-nm --numeric-sort --format=just-symbols - | FileCheck %s --check-prefix=ORDERFILE
```

Comment:
I think these lines and all the following lines should go at the top of the file. Anything after the `clang` command should be generated by that command.

> The script will prepare extra files with split-file, invoke gen, and then rewrite the part after gen with its stdout.

https://llvm.org/docs/TestingGuide.html#elaborated-tests